daemon/api.go

```diff
@@ -327,6 +332,22 @@ func getSnapsInfo(c *Command, r *http.Request) Response {
 		}
 	}
 
+	if len(includeTypes) > 0 {
+		for i, r := range results {
+			hasType := false
```

Comment:
Another good point! I was concerned about how long the `getSnapsInfo` function was getting so perhaps we can improve both readability and maintainability in one go by extracting the type filtering into it's own function.